### PR TITLE
Functional test

### PR DESCRIPTION
**Pull request by tarekziade**
## Closes https://github.com/elastic/enterprise-search-team/issues/2386

Adds `make ftest`

That target will:

- create a stack in Docker (Mongo+Elasticsearch)
- add sample data in mongo (airbnb)
- run a full sync
- verify that Elasticsearch received the documents
- tear down the stack

in a follow-up issue we'll have it running once a day in Jenkins

### CODE REVIEW

File: tests/ftest.rb [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+end
+
+def run_sync
+  # late import because the lib import the config when loaded
+  # and we change it in `set_auth`
+  $LOAD_PATH << LIB_DIR
+
+  Dir["#{LIB_DIR}/**/*.rb"].sort.each { |f|
+    next if f.include?('lib/app')
+    next if f.include?('connectors_async')
```
Comment (wangch079) [inline tests/ftest.rb:73]: Think we don't have this anymore

File: tests/ftest.rb [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    f = File.join(relative_dir, name)
+    require(f)
+  }
+
+  require 'active_support/json'
+  require 'core/elastic_connector_actions'
+  require 'core/connector_settings'
+
+  puts('Create the connector')
+  Core::ElasticConnectorActions.ensure_connectors_index_exists
```
Comment (wangch079) [inline tests/ftest.rb:87]:
```suggestion
  Core::ElasticConnectorActions.ensure_connectors_index_exists
  Core::ElasticConnectorActions.ensure_job_index_exists
```

File: tests/ftest.rb [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+  connector_id = Core::ElasticConnectorActions.create_connector('mongo', 'mongo')
```
Comment (wangch079) [inline tests/ftest.rb:88]: I'm currently [refactoring the CLI](https://github.com/elastic/connectors-ruby/pull/224), and the content index should be created in `create_connector`.

But that is at the bottom of my TO-DO list. I will update the test if it's merged before my refactor PR.

File: tests/ftest.rb [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+
+  config_settings = Core::ConnectorSettings.fetch(connector_id)
+
+  # Creating the mapping for the index
+  settings = Utility::Elasticsearch::Index::TextAnalysisSettings.new(:language_code => 'en', :analysis_icu => false).to_h
+  mappings = Utility::Elasticsearch::Index::Mappings.default_text_fields_mappings(:connectors_index => true)
+  body_payload = { settings: settings, mappings: mappings }
+  Core::ElasticConnectorActions.ensure_index_exists(config_settings[:index_name], body_payload)
+
+  Core::SyncJobRunner.new(config_settings, App::Config[:service_type]).execute
```
Comment (wangch079) [inline tests/ftest.rb:115]: Should set `sync_now = true` before run the `SyncJobRunner`

File: tests/ftest.rb [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+  raise StandardError.new "Bad Count #{count}" unless count > 5000
+  puts('Verified!')
+end
+
+if __FILE__ == $PROGRAM_NAME
+  start_stack
+  prepare_es
+  set_auth
+  run_sync
+  verify
```
Comment (wangch079) [inline tests/ftest.rb:132]: There should be a cleanup step?

File: tests/ftest.rb (hunk 3 above)
Comment (tarekziade) [reply tests/ftest.rb:89]: sounds good, thanks for the heads up

File: tests/ftest.rb (hunk 5 above)
Comment (tarekziade) [reply tests/ftest.rb:133]: good point! adding it -- I will simply teardown the stack

File: tests/ftest.rb (hunk 4 above)
Comment (tarekziade) [reply tests/ftest.rb:126]: where should this be added?

File: tests/ftest.rb (hunk 1 above)
Comment (tarekziade) [reply tests/ftest.rb:83]: What about https://github.com/elastic/connectors-ruby/blob/main/lib/connectors_async.rb ?

File: tests/ftest.rb (hunk 4 above)
Comment (wangch079) [reply tests/ftest.rb:125]:
```
Core::ElasticConnectorActions.force_sync(connector_id)
```

**Review by wangch079 — APPROVED** [review]

**Comment by artem-shelkovnikov** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.4|[](https://github.com/elastic/connectors-ruby/pull/255)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**